src/Microsoft.AspNetCore.Authentication/Internal/AuthenticationEventSource.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Diagnostics.Tracing;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Authentication.Internal
+{
+    [EventSource(Name = "Microsoft-AspNetCore-Authentication")]
+    public class AuthenticationEventSource : EventSource
+    {
+        public static readonly AuthenticationEventSource Log = new AuthenticationEventSource();
+        private readonly EventCounter _authenticationMiddlewareDuration;
+
+        private AuthenticationEventSource()
+        {
+            _authenticationMiddlewareDuration = new EventCounter("AuthenticationMiddlewareDuration", this);
+        }
+
+        [NonEvent]
+        internal EventSpan<AuthenticationEventSource> AuthenticationMiddlewareStart(string traceIdentifier, PathString path)
+        {
+            if (IsEnabled(EventLevel.Informational, EventKeywords.None))
+            {
+                AuthenticationMiddlewareStart(traceIdentifier, path.Value);
+            }
+
+            return EventSpan.Create(this, (self, duration) => self.AuthenticationMiddlewareEnd(traceIdentifier, path, duration));
```

Comment:
Allocating a closer per event isn't promising.

Reply:
Yeah. This needs revision for sure.